### PR TITLE
Spring upgrade

### PR DESCRIPTION
**Pull request by irinaepshteyn**

### CODE REVIEW

File: Jenkinsfile
Comment (anny1099) [inline Jenkinsfile:21]: checked spring-filters-config, set-env-metering-filter.sh is different in fixMeteringIntegTests vs. master. not sure why switch to master.

**Review by anny1099 — CHANGES_REQUESTED** [review]

File: Jenkinsfile
Comment (metacubed) [inline Jenkinsfile:21]: I remember Pooja had a specific commit on the "fixMeteringTests" branch. Some tests failed without that commit on the "master" branch.

File: pom.xml
Comment (metacubed) [inline pom.xml]: Change commons-lang to commons-lang3, there's a package change.
Also commons-collections to commons-collections4.

File: Jenkinsfile
Comment (irinaepshteyn) [reply Jenkinsfile:21]: @anny1099, I merged changes to set-env-metering-filter.sh into master.  There is no reason to keep these changes on a branch.

**Review by metacubed — CHANGES_REQUESTED** [review]

File: Jenkinsfile
Comment (irinaepshteyn) [reply Jenkinsfile:21]: @metacubed, I merged changes to set-env-metering-filter.sh into master. There is no reason to keep these changes on a branch.

Comment (anny1099) [reply Jenkinsfile:21]: there are assertion steps in the IT test we commented out in the previous story, due to nurego API performance change, if you switch branch for testing, you may want to make sure the original IT test (non commented out version) still works at least locally. 
also, if we decides to switch branch for IT testing, the ReadMe file should update.

Comment (irinaepshteyn) [reply Jenkinsfile:21]: I merged changes to set-env-metering-filter.sh into master because these changes are prerequisites for the tests to start.  These changes have nothing to do with the commented assertions and keeping them on a branch doesn't make sense.  It just creates confusion.

I agree with changing the README.

File: pom.xml
Comment (irinaepshteyn) [reply pom.xml]: @metacubed, the only method used from apache commons StringUtils is isNotEmpty().  I am thinking of changing it to spring framework StringUtils hasText().  One less package to maintain.  What do you think?

Comment (irinaepshteyn) [reply pom.xml]: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/StringUtils.html#hasText-java.lang.String-

http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotEmpty-java.lang.CharSequence-

Both return true if the String is not null, its length is greater than 0, and it contains at least one non-whitespace character.

File: Jenkinsfile
Comment (anny1099) [reply Jenkinsfile:21]: agreed that set-env-metering-filter.sh can be merged into master, also I assume you will create a separate PR for the merging in spring-filters-config? btw, I think that change should be in first before this PR.

File: pom.xml
Comment (metacubed) [reply pom.xml]: Sounds good, @irinaepshteyn.

File: Jenkinsfile
Comment (irinaepshteyn) [reply Jenkinsfile:21]: @anny1099, this merge is already done.

**Comment by irinaepshteyn** [comment]
@metacubed, @anny1099, could you review once again and approve if agree.

**Review by anny1099 — APPROVED** [review]
lgtm

**Review by metacubed — APPROVED** [review]

**Pull request merged**
